storage/rocksdb/ha_rocksdb.cc

```diff
@@ -3863,6 +4020,82 @@ static bool rocksdb_show_status(handlerton *const hton, THD *const thd,
   return res;
 }
 
+/*
+  Implements Log_resource lock.
+
+  returns false on success
+*/
+static bool rocksdb_lock_hton_log(handlerton *const /* unused */) {
+  DBUG_ASSERT(rdb != nullptr);
+  return !rdb->LockWAL().ok();
+}
+
+/*
+  Implements Log_resource unlock.
+
+  returns false on success
+*/
+static bool rocksdb_unlock_hton_log(handlerton *const /* unused */) {
+  DBUG_ASSERT(rdb != nullptr);
+  return !rdb->UnlockWAL().ok();
+}
+
+/*
+  Implements Log_resource collect_info.
+
+  Produces JSON object with following structure:
+
+  "RocksDB": {
+    "wal_files": [
+      { "log_number": N, "path_name": "...", "file_size_bytes": K },
```

Comment:
do you guys think that {"number": N, "path", "...", "size": K} would sound better? I just copied RocksDB method names here, but we don't really need to follow their naming...

Reply:
I think this is fine as-is.